### PR TITLE
make pattern_* and makeiplist tools faster

### PR DESCRIPTION
**Pull request by Auxilus**
before:
```
time ./pattern_create.rb -l 50 -s ABC,def,123
Ad1Ad2Ad3Ae1Ae2Ae3Af1Af2Af3Bd1Bd2Bd3Be1Be2Be3Bf1Bf

________________________________________________________
Executed in   13.77 secs   fish           external
   usr time    7.37 secs    0.61 millis    7.37 secs
   sys time    6.33 secs    1.61 millis    6.33 secs
``` 
```
time ./pattern_offset.rb -q Aa3A
[*] Exact match at offset 9

________________________________________________________
Executed in   13.38 secs   fish           external
   usr time    7.21 secs  774.00 micros    7.21 secs
   sys time    6.09 secs  627.00 micros    6.09 secs
```
```
time ./makeiplist.rb -h
This script takes a list of ranges and converts it to a per line IP list.
Usage: ./makeiplist.rb [options]

Specific options:
    -i <filename>                    Input file
    -o <filename>                    (Optional) Output file. Default: iplist.txt

Common options:
    -h, --help                       Show this message

________________________________________________________
Executed in   13.31 secs   fish           external
   usr time    7.32 secs    0.00 millis    7.32 secs
   sys time    5.91 secs    2.72 millis    5.91 secs
```

After:
```
time ./pattern_create.rb -l 50 -s ABC,def,123
Ad1Ad2Ad3Ae1Ae2Ae3Af1Af2Af3Bd1Bd2Bd3Be1Be2Be3Bf1Bf

________________________________________________________
Executed in  363.86 millis    fish           external
   usr time  266.98 millis    0.00 micros  266.98 millis
   sys time   96.74 millis  1108.00 micros   95.63 millis
```

```
time ./pattern_offset.rb -q Aa3A
[*] Exact match at offset 9

________________________________________________________
Executed in  396.77 millis    fish           external
   usr time  300.23 millis  612.00 micros  299.62 millis
   sys time   96.39 millis  507.00 micros   95.88 millis
```

```
 time ./makeiplist.rb -h
This script takes a list of ranges and converts it to a per line IP list.
Usage: ./makeiplist.rb [options]

Specific options:
    -i <filename>                    Input file
    -o <filename>                    (Optional) Output file. Default: iplist.txt

Common options:
    -h, --help                       Show this message

________________________________________________________
Executed in    6.98 secs   fish           external
   usr time    4.28 secs    9.76 millis    4.27 secs
   sys time    2.71 secs    1.32 millis    2.71 secs
```

## Testing

Verify there are no explosions:

- `./tools/exploit/pattern_create.rb -l 50 -s ABC,def,123`
- `./tools/exploit/pattern_offset.rb -q Aa3A`
- `tools/exploit/nasm_shell.rb`
- `./tools/recon/makeiplist.rb  --help`

### CODE REVIEW

File: tools/exploit/nasm_shell.rb [hunk 1]
```diff
@@ -17,8 +17,6 @@
 end
 
 $:.unshift(File.expand_path(File.join(File.dirname(msfbase), '..', '..', 'lib')))
-require 'msfenv'
```
Comment (wvu-r7) [inline tools/exploit/nasm_shell.rb]: This breaks the tool:

```
wvu@kharak:/rapid7/metasploit-framework:master$ tools/exploit/nasm_shell.rb
Traceback (most recent call last):
	2: from tools/exploit/nasm_shell.rb:42:in `<main>'
	1: from tools/exploit/nasm_shell.rb:42:in `new'
/rapid7/metasploit-framework/lib/rex/ui/text/output/stdio.rb:36:in `initialize': undefined method `assert_valid_keys' for {}:Hash (NoMethodError)
wvu@kharak:/rapid7/metasploit-framework:master$
```

Was this tested?

Comment (Auxilus) [reply tools/exploit/nasm_shell.rb]: yes, sorry, forgot to restore msfenv 😅

**Comment by wvu-r7** [comment]
This has been transferred to the Belfast team. Thanks.

File: tools/exploit/nasm_shell.rb (hunk 1 above)
Comment (adfoster-r7) [reply tools/exploit/nasm_shell.rb]: Seems like this `Rex::Ui::Text::Output::Stdio` is depending on `assert_valid_keys` which actually comes from activesupport. This seems like something that could be fixed without much effort, but happy to leave this to a separate PR :+1:

**Comment by adfoster-r7** [comment]
## Release Notes

The pattern_create, pattern_offset, and makeiplist tools now load much faster. The pattern tools in particular are down from 6-7 seconds to 0-1 seconds.

**Pull request merged**